### DISCUSSION TITLE
Data smoothing

### DISCUSSION THREAD

**Discussion by ghost** [atoti/atoti#705] [Q&A] [answered]
Hello,
is it possible to add a way to "smooth" the data from the GUI by proposing some MA / EMA / EWMA for example, on different horizons ? We could obviously do it on the input data but having a way to do it dynamically with some clicks would be very helpful !
Thanks

**Comment by mailys-ds** [comment]
Hello,
In the GUI, you can dynamically implement MA/EMA/EWMA easily in few clicks with the calculated measure feature (calculated measure -> Evolution ) as showed in this video:
[Screencast from 10-05-2022 12:48:21 PM.webm](https://user-images.githubusercontent.com/62394323/194044004-a09f1fbd-eb30-423b-9127-c8642461e474.webm).
Let me know it that works for you.
Thanks.

**Comment by ghost** [comment]
Hi Mailys,

I don't see the "Fx" button in our GUI unfortunately, does that require a more recent version ? (even our logo at the top left is different).
But otherwise the feature you propose seems to answer our need, so I am willing to go to that version if available !
We use version 0.7.0 at the moment, which is the latest available to download I believe..

Thanks

**Comment by mailys-ds** [comment, marked as answer]
Hi, 
It should work on atoti 0.7.0, however the _CalculatedMeasure_ feature is not enabled by default, that's why you can't see it in the GUI. You can enable it by setting the `canUseCalculatedMeasures` permission to true. You also need to ensure that you have a CalculatedMeasure folder under /ui in the setting. 
To do that, access the dashboard application with `session.link()` from a notebook, add `/admin` to the URL to go to the settings and then go to `user_roles` -> `permission` and add `"canUseCalculatedMeasures": true`. (See gif)
![enableCalculatedMeasure gif](https://user-images.githubusercontent.com/62394323/196972201-d3908f06-c106-4c6a-8809-f8066402f3ae.gif)

Coming back to your application dashboard, you hsould see the "Fx" button.

**Reply by ghost** [reply]
Hi Mailys,

We don't have the `user_roles` section in the admin configuration. Is it part of the atoti-plus plugin ?

![image](https://user-images.githubusercontent.com/102600205/197002972-b8900610-1433-4560-8702-78715c24a08c.png)

Thanks

**Reply by mailys-ds** [reply]
Upgrading atoti to 0.7.1 should enable you to access the permission file as the UI has been upgraded in atoti 0.7.1 and is now generating the permission file by default. Otherwise adding a permission file manually should work wit hatoti 0.7.0.
